### PR TITLE
[WIP] Input: clear all handlers whenever the pad_thread is disabled

### LINKED ISSUE #5985: DS4 Handler crashes after connecting controller mid-emulation
1. Start gs_gcm_cube.elf or anything with DS4 selected and a disconnected PS4 Pad
2. Turn on your DS4 after booting the game
3. Open pads in the middle of the game and press refresh until DS4 Pad shows as connected
4. Save changes
5. RPCS3 will hard crash (Unhandled exception at 0x000007FEFBBF1454 in rpcs3.exe: 0xC0000005: Access violation executing location 0x000007FEFBBF1454.)

Crashes on
https://github.com/RPCS3/rpcs3/blob/master/rpcs3/Input/ds4_pad_handler.cpp#L786
on
https://github.com/RPCS3/hidapi/blob/master/windows/hid.c#L393

Tested on Windows 7, current master (https://github.com/RPCS3/rpcs3/commit/4037225e989ec798593d64e0ac1a1f887469d11e) and with an official DS4 controller

### PR DESCRIPTION
**Pull request by Megamouse**
Please test this with all the handlers

fixes #5985

### CODE REVIEW

File: rpcs3/rpcs3qt/game_compatibility.cpp [hunk 1]
```diff
@@ -127,7 +127,7 @@ void game_compatibility::RequestCompatibility(bool online)
 	QNetworkReply* network_reply = m_network_access_manager->get(m_network_request);
 
 	// Show Progress
-	m_progress_dialog.reset(new progress_dialog(tr(".Please wait."), tr("Abort"), 0, 100));
+	m_progress_dialog = new progress_dialog(tr(".Please wait."), tr("Abort"), 0, 100);
```
Comment (elad335) [inline rpcs3/rpcs3qt/game_compatibility.cpp]: Won't it cause a memory leak when there are multiple requests for the compat database at the same run? I think using unique_ptr was correct.

Comment (Megamouse) [reply rpcs3/rpcs3qt/game_compatibility.cpp]: Idk. Doesn't deleteLater take care of that?

Comment (elad335) [reply rpcs3/rpcs3qt/game_compatibility.cpp]: I didn't see any delete statement which is why I was asking. looking at the code again there's none.

Comment (elad335) [reply rpcs3/rpcs3qt/game_compatibility.cpp]: Only an explicit call to delete operator deallocates a raw pointer.

Comment (Megamouse) [reply rpcs3/rpcs3qt/game_compatibility.cpp]: Dialog::Close, m_pdlg, dialog::deleteLater

Comment (elad335) [reply rpcs3/rpcs3qt/game_compatibility.cpp]: ahh, seems fine then.

**Comment by AniLeo** [comment]
#5985 still happens with this PR

**Comment by Megamouse** [comment]
Closing this because it doesn't work anyway

**Pull request closed without merging**